fg/apps/main/templates/maps/orders.html

```diff
@@ -1,64 +1,74 @@
 {% extends "base/page.html" %}
 {% load staticfiles %f}
+
 {% block content %}
+  {% include "messages/message.html" %}
+  <div class="container" style="padding-top: 100px;">
+    <h1>Orders Map</h1>
+    <div id='map' style='width: 800px; height: 600px;'></div>
+  </div>
+{% endblock %}
 
+{% block pagescripts %}
 <style>
-.node rect {
-  cursor: pointer;
-  fill: #fff;
-  fill-opacity: 0.5;
-  stroke: #3182bd;
-  stroke-width: 1.5px;
-}
-
-.node text {
-  font: 10px sans-serif;
-  pointer-events: none;
-}
+  .marker {
+    background-image: url('https://docs.mapbox.com/help/data/marker-editor.svg');
+    background-size: cover;
+    width: 25px;
+    height: 25px;
+    border-radius: 50%;
+    cursor: pointer;
+  }
+  .mapboxgl-popup {
+    max-width: 200px;
+  }
 
-.link {
-  fill: none;
-  stroke: #9ecae1;
-  stroke-width: 1.5px;
-}
+  .mapboxgl-popup-content {
+    text-align: center;
+    font-family: 'Open Sans', sans-serif;
+  }
 </style>
-
-{% include "messages/message.html" %}
-<div class="container">
-   <h1>Orders Map</h1>
-   <hr>
-   <div id="map" style="position: relative; width: 800px; height: 600px;"></div>
-</div>
-{% endblock %}
-{% block pagescripts %}
-<script src="//cdnjs.cloudflare.com/ajax/libs/d3/3.5.3/d3.min.js"></script>
-<script src="//cdnjs.cloudflare.com/ajax/libs/topojson/1.6.9/topojson.min.js"></script>
-<script src="//cdnjs.cloudflare.com/ajax/libs/datamaps/0.5.9/datamaps.world.min.js"></script>
+<script src='https://api.tiles.mapbox.com/mapbox-gl-js/v1.3.1/mapbox-gl.js'></script>
+<link href='https://api.tiles.mapbox.com/mapbox-gl-js/v1.3.1/mapbox-gl.css' rel='stylesheet' />
 <script>
+  mapboxgl.accessToken = 'pk.eyJ1Ijoic2hlYTI1NiIsImEiOiJjazB5dXJqYTgwN3p5M2NvODhrYWY4ejBuIn0.WYrEJAAKQBR3i38SU3nllQ';
+
```

Comment:
This looks great! Before I test, could you please:

 - confirm the [scopes](https://docs.mapbox.com/help/how-mapbox-works/access-tokens/) of the token that you've created
 - have the token provided by the user in the settings/config.py and add to the view - we would never want to require the user to edit the view (html) to update it.
 - add documentation to the map docs to tell the user about how the map is generated, link to the mapbox, and instruct how to generate, scope, and update the token (the one provided in the config.py settings should be faux (e.g.) `MAPBOX_TOKEN=xxxxxxxxxxxx`)
 - If you are adding metadata about an order, (right now all are hard coded as "OpenEnzyme" and a dummy description) take a look at how we use the Shippo API to obtain it, and make a suggestion (even if you can't test) for how we can update the data extraction function to obtain what is needed. If you update this (fg/apps/main/management/commands/generate_mapdata.py) I can test it for you on the server, and provide an updated ordercoords.json to test from.

 - update the screen shot in the docs to be the current map
This will be a general pattern that we want to follow - if something is added that warrants a token (or the user to understand something) we typically want to add it to the docs (in the docs/_docs) folder. There is already a section for the map, so you should be able to update it.

I'll be able to preview/test this out when we get some of the current updates finished up. Thanks for working on this!

Reply:
oh and this is hugely important - (and you actually reminded me in your issue comment!) We need a map that supports markers that can be scaled to the number of orders (akin to the current) or a heatmap style that does similar. What we can't do is have individual markers for each order - that kind of map is more appropriate for locating single entities, but not representing size/quantity. I don't make a ton of maps, but if you look at the  [code here](http://vsoch.github.io/bodymap/) when you click map it supports zooming too, and if I remember correctly you can use that map style for more than just individual points.

Reply:
I was going to start with that to refactor the map (and address this issue) - of course not styling so dark :P That particular dataset I found randomly and is fairly dark, I just made the map with the body for fun. But it's dark.

Reply:
And to answer your question - yes we want to eventually have filters based on order types, so keep that in mind as you design. Okay back to work!

Reply:
Hey thanks @vsoch! Please excuse the delay. Please find my responses inline below:

> confirm the scopes of the token that you've created

This is actually the default public token so it is the same for everyone. I will document this.

> have the token provided by the user in the settings/config.py and add to the view - we would never want to require the user to edit the view (html) to update it.

Will do. I will set the default token for the user, and if they want to change it to a registered token they can.

> add documentation to the map docs to tell the user about how the map is generated, link to the mapbox, and instruct how to generate, scope, and update the token (the one provided in the config.py settings should be faux (e.g.) MAPBOX_TOKEN=xxxxxxxxxxxx)

Will do, thanks. Didn't want to do this until getting some level of feedback from you, just in case I had to go down a different route.

> If you are adding metadata about an order, (right now all are hard coded as "OpenEnzyme" and a dummy description) take a look at how we use the Shippo API to obtain it, and make a suggestion (even if you can't test) for how we can update the data extraction function to obtain what is needed. If you update this (fg/apps/main/management/commands/generate_mapdata.py) I can test it for you on the server, and provide an updated ordercoords.json to test from.

OK this is exactly what I wanted to do, but was unable to run that command. It would be fairly hard to accomplish this if I have to rely on you to generate the `ordercoords.json` file. Is there documentation on how I can run it? If so, I can handle the additional data tagging.

> update the screen shot in the docs to be the current map
This will be a general pattern that we want to follow - if something is added that warrants a token (or the user to understand something) we typically want to add it to the docs (in the docs/_docs) folder. There is already a section for the map, so you should be able to update it.

Sure thing!

> oh and this is hugely important - (and you actually reminded me in your issue comment!) We need a map that supports markers that can be scaled to the number of orders (akin to the current) or a heatmap style that does similar. What we can't do is have individual markers for each order - that kind of map is more appropriate for locating single entities, but not representing size/quantity. 

Sounds good.

> And to answer your question - yes we want to eventually have filters based on order types, so keep that in mind as you design. Okay back to work!

Makes sense. Will keep that in mind so that we can add an extension to the design in the future.

Thanks!

Reply:
> This is actually the default public token so it is the same for everyone. I will document this.

Oh that's really interesting! I've never seen a shared API token - I wonder how they track usage per client? Yes, please add to the documentation, and include limits (I'm guessing there are some) and how to update the token (in the settings/config.py) if the user wants to get a registered token.

> OK this is exactly what I wanted to do, but was unable to run that command. It would be fairly hard to accomplish this if I have to rely on you to generate the ordercoords.json file. Is there documentation on how I can run it? If so, I can handle the additional data tagging.

You would need a Shippo API token - and actually not a test token, but a production token, and I don't think I can give this to you, but I can get the data you need from the API for you. Here is what I just ran for you (a subset of the file, manually):

```python
from fg.apps.main.utils import load_json
from fg.settings import SHIPPO_TOKEN

import os
import json
import requests
import shutil
import sys

url="https://api.goshippo.com/orders/"
headers={"Authorization": "ShippoToken %s" % SHIPPO_TOKEN}

response = requests.get(url, headers=headers)
listing = []

while response.status_code == 200:

            results = response.json()
            if "results" not in results:
                print("No results in response.")
                sys.exit(1)

            listing = listing + results['results']
            response = requests.get(results['next'], headers=headers)
```
I've dumped to json, and I can send this to you through your private email with names removed (please send this to me).

This script is typically run with `python manage.py generate_mapdata` but I would recommend loading the data that I send you and doing testing manually. [Here (https://github.com/vsoch/freegenes/blob/master/fg/apps/main/management/commands/generate_mapdata.py) is a direct link to the script and it's run via [this cron job](https://github.com/vsoch/freegenes/blob/master/Dockerfile#L58) that is created on the container build. Per your request, I've updated the [orders map](https://vsoch.github.io/freegenes/docs/usage/interface#orders-map) with this complete information.

Thanks @shea256 !

Reply:
@vsoch Submitted a request for review. Let me know when you've gotten a chance to review and whether you give it the green light. Thanks!